backend/backend.go

```diff
@@ -0,0 +1,59 @@
+package main
+
+import (
+	"log"
+	"net/http"
+	"path/filepath"
+	"strconv"
+
+	"gopkg.in/ini.v1"
+
+	"github.com/ucsdscheduleplanner/UCSD-Schedule-Planner/backend/db"
+	"github.com/ucsdscheduleplanner/UCSD-Schedule-Planner/backend/routes"
+)
+
+// TODO: make this config-able
+const port = 8080
+
+// create closure for http handler func
+func makeHandler(f func(http.ResponseWriter, *http.Request, *db.DatabaseStruct), ds *db.DatabaseStruct) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) { f(w, r, ds) }
+}
+
+func main() {
+
+	var config, err = ini.Load(filepath.Join(".", "config", "config.example.ini"))
+
+	// var config, err = ini.Load(filepath.Join(".", "config", "config.dev.ini"))
+
+	if err != nil {
+		panic("Error reading config file: " + err.Error())
+	}
+
+	if config == nil {
+		panic("Could not find configuration file, shutting down")
+	}
+
+	// open once, close once
+	ds, err := db.NewIni(config)
+	if err != nil {
+		panic("Failed to init db: " + err.Error())
+	}
+
+	// only for the completeness of the code
+	defer ds.Close()
```

Comment:
move this defer statement right after initialization just to be safe, an error could have occurred but the thing was still initialized so that's a memory leak

Reply:
Sure I will change but not sure if close an uninitialized thing will be an error or not.

https://github.com/GoogleCloudPlatform/golang-samples/blob/a699cc86997175aa096113486151599df8948a10/firestore/firestore_snippets/main.go#L53

Reply:
https://github.com/search?q=sql.open+close+language%3Agolang&type=Code